src/app/pages/home/update-contact-dialog/update-contact-dialog.component.ts

```diff
@@ -44,8 +53,13 @@ export class UpdateContactDialogComponent {
     const data = this.contactForm.getRawValue();
     try {
       await this.contactProvider.updateContact(data);
+      this.onChange.emit();
+      this.snackbarService.successMessage("Contato alterado com sucesso")
+      this.dialogRef.close();
     } catch (error: any) {
-      console.log(error);
+      if (error?.response?.data?.errors) {
```

Comment:
não me agradou mt esse trecho de código, o ideal seria vc capitar a resposta da Api e verificar se o length da propriedade 'errors' da resposta é maior que zero, se for, significa que a api encontrou um erro durante o processamento da requisição. Então esses erros devem ser exibidos com aquele component generico de tratamento de mensagens de erro.

o ideal seria que nesse bloco try/catch, dentro do try ficasse a chamada da api e as ações necessárias a serem tomadas após a resposta da api e dentro do catch ficar o tratamento de erro para o caso de o backend 'quebrar' durante o tratamento da requisição, nessa caso tbm poderia ser exibida uma mensagem de erro generica como 'Ocorreu um erro ao salvar o contato, tente novamente.' usando o componente genérico que foi criado.

caso tenha dúvidas podemos ver esse ponto em conjunto 👍

Reply:
isso vale para todos os códigos similares a esse em outros componentes

Reply:
seguir o comentário feito no component de criação de contato